tests/foreman/performance/common/perf_candlepin.py

```diff
@@ -0,0 +1,94 @@
+from robottelo.common import conf, ssh
+
+import logging
+import time
+import requests
+
+class CandlepinPerfTest:
+    """
+    Measures performance of RH Satellite 6 - Candlepin Subscription functionality
+    """
+    username = conf.properties['foreman.admin.username']
+    password = conf.properties['foreman.admin.password']
+    serverhost = conf.properties['main.server.hostname']
+
+    # create logger
+    logger = logging.getLogger('robottelo')
+    logger.setLevel(logging.DEBUG)
+    requests.packages.urllib3.disable_warnings()
+
+    '''
+    Subscribe VM to Satellite by Register + ActivationKey
+    '''
+    @classmethod
+    def single_register_activation_key(cls, ak_name, default_org, vm):
+        # note: must create ssh keys for vm if running on local
+        # result = ssh.command('subscription-manager clean', hostname=vm)
+        result = ssh.command('subscription-manager clean', hostname=vm)
+        if result.return_code != 0:
+           cls.logger.error('Failed to clean sub-mgr!')
+           cls.logger.error(result.stderr)
+
+        result = ssh.command('time -p subscription-manager register --activationkey={0} --org={1}'.format(ak_name, default_org), hostname=vm)
+        # if result.return_code == 0:
+        real_time = [real for real in result.stderr.split('\n') if real.startswith('real')]   
+        real_time = real_time[0].split(' ')[1]
+        return float(real_time)
+
+    '''
+    Subscribe VM to Satellite by Register + Attach
+    '''
+    @classmethod
+    def single_register_attach(cls, sub_id, default_org, environment, vm):
+        result = ssh.command('subscription-manager clean', hostname=vm)
+        time_reg = cls.sub_mgr_register_authentication(default_org, environment, vm)
+        # pool_id = cls.get_subscription_pool_id()
+        time_att = cls.sub_mgr_attach(sub_id, vm)
+        
+        if time_reg == None or time_att == None:
+            total_time = 0
+        else:
+            total_time = time_reg+time_att
+        return time_reg, time_att, total_time
+
+    '''
+    Helper Function: subscription-manager register -u -p --org --environment
+    '''
+    @classmethod
+    def sub_mgr_register_authentication(cls, default_org, environment, vm):
+        # result = ssh.command('subscription-manager clean', hostname=cls.vm)
+        result = ssh.command('time -p subscription-manager register --username={0} --password={1} --org={2} --environment={3}'.format(cls.username, cls.password, default_org, environment), hostname=vm)
```

Comment:
Wrap code. Also, @elyezer [has mentioned](https://github.com/SatelliteQE/robottelo/pull/2522#discussion-diff-33732511) that using `time` can break CLI automation if not done right. Take that in to consideration.